addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -317,13 +318,29 @@ options.registry.gallery = options.Class.extend({
                     imgs.push(data.$image[0]);
                     break;
             }
+            position = _.indexOf(imgs, data.$image[0]);
             _.each(imgs, function (img, index) {
                 // Note: there might be more efficient ways to do that but it is
                 // more simple this way and allows compatibility with 10.0 where
                 // indexes were not the same as positions.
                 $(img).attr('data-index', index);
             });
-            this.mode('reset', this.getMode());
+            const currentMode = this.getMode();
+            this.mode('reset', currentMode);
+            if (currentMode === 'slideshow') {
+                const $carousel = this.$target.find('.carousel');
+                $carousel.removeClass('slide');
+                $carousel.carousel(position);
+                this.$target.find('.carousel-indicators li').removeClass('active').addClass('false');
+                this.$target.find('.carousel-indicators li[data-slide-to="' + position + '"]').removeClass('false').addClass('active');
+                this.trigger_up('activate_snippet', {
+                    $snippet: this.$target.find('.carousel-item.active img'),
+                    ifInactiveOptions: true,
+                });
+                $carousel.addClass('slide');
+            } else {
+                data.$image[0].click();
```

Comment:
Can't you know which image to activate without the need to search for it in slideshow mode and simulate a click on it otherwise? Can't you do something like that in both cases?
```javascript
this.trigger_up('activate_snippet', {
    $snippet: data.$image.eq(position),
    ifInactiveOptions: true,
});
```
Not sure to understand the code here.